src/test/java/core/basesyntax/bookstore/service/CategoryServiceTest.java

```diff
@@ -0,0 +1,165 @@
+package core.basesyntax.bookstore.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
+import core.basesyntax.bookstore.dto.category.CategoryDto;
+import core.basesyntax.bookstore.dto.category.CreateCategoryRequestDto;
+import core.basesyntax.bookstore.dto.category.UpdateCategoryRequestDto;
+import core.basesyntax.bookstore.exception.EntityNotFoundException;
+import core.basesyntax.bookstore.mapper.CategoryMapper;
+import core.basesyntax.bookstore.model.Category;
+import core.basesyntax.bookstore.repository.category.CategoryRepository;
+import core.basesyntax.bookstore.service.impl.CategoryServiceImpl;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.Pageable;
+
+@ExtendWith(MockitoExtension.class)
+public class CategoryServiceTest {
+    private static final Long ID_SCI_FI = 1L;
+    private static final String NAME_SCI_FI = "sci-fi";
+    private static final Category SCI_FI = new Category(ID_SCI_FI);
+    private static final Long ID_ACTION = 2L;
+    private static final String NAME_ACTION = "action";
+    private static final Category ACTION = new Category(ID_ACTION);
+
+    private static final CategoryDto sciFiDto = new CategoryDto(ID_SCI_FI, NAME_SCI_FI, null);
+    private static final CategoryDto actionDto = new CategoryDto(ID_ACTION, NAME_ACTION, null);
+
+    @Mock
+    private CategoryRepository categoryRepository;
+
+    @Mock
+    private CategoryMapper categoryMapper;
+
+    @InjectMocks
+    private CategoryServiceImpl categoryService;
+
+    @BeforeAll
+    static void beforeAll() {
```

Comment:
be attentive, some test cases can change objects' state

Reply:
I keep track of them to not to